### PR TITLE
Add basic bench testing for analyzers

### PR DESCRIPTION
**Pull request by sushicw**
Add basic performance bench tests for analyzers. For https://github.com/istio/istio/issues/18008. 

These come in two forms:

* A simple bench test using our unit test case data (minimal input sizes). This is simply a benchmarked version of the required analyzer unit tests.  
* A bench test using artificially constructed test data, with N of every possible resource (and each resource being entirely blank). 

This doesn't give anywhere near perfect coverage and neither scenario is really "realistic", but it is enough to highlight some analyzers as being outliers in terms of how long they take to run, and that's something we can take action on. 

Example output (filtering out results with < 0.01 ns/op):
```
 crwilson@crwilson:~/go/src/istio.io/istio$ go test -bench=. -run=XXX ./galley/pkg/config/analysis/analyzers -v | grep -v "0\.00"
goos: linux
goarch: amd64
pkg: istio.io/istio/galley/pkg/config/analysis/analyzers
BenchmarkAnalyzers/virtualServiceDestinationHosts-bench-8                        	1000000000	         0.0115 ns/op
BenchmarkAnalyzersArtificialBlankData100/gateway.IngressGatewayPortAnalyzer-bench-8       	1000000000	         0.0231 ns/op
BenchmarkAnalyzersArtificialBlankData200/gateway.IngressGatewayPortAnalyzer-bench-8                         	1000000000	         0.181 ns/op
BenchmarkAnalyzersArtificialBlankData400/gateway.IngressGatewayPortAnalyzer-bench-8                         	       1	1436830870 ns/op
BenchmarkAnalyzersArtificialBlankData800/auth.MTLSAnalyzer-bench-8                                          	1000000000	         0.0216 ns/op
BenchmarkAnalyzersArtificialBlankData800/gateway.IngressGatewayPortAnalyzer-bench-8                         	       1	11516476224 ns/op
PASS
ok  	istio.io/istio/galley/pkg/config/analysis/analyzers	22.562s
```
Note that bench results can be further profiled, see https://medium.com/@felipedutratine/profile-your-benchmark-with-pprof-fb7070ee1a94 for one article about how.

### CODE REVIEW

**Comment by howardjohn** [comment]
See https://github.com/istio/istio/issues/20123 which was inspired by this, may be something possible to leverage by this long term. Its pretty nebulous future though

File: galley/pkg/config/analysis/analyzers/analyzers_test.go [hunk 1]
```diff
@@ -506,6 +475,53 @@ func TestAnalyzersHaveDescription(t *testing.T) {
 	}
 }
 
+func setupAndRunCase(tc testCase, cr snapshotter.CollectionReporterFn) (local.AnalysisResult, error) {
+	// Default processing log level is too chatty for these tests
```
Comment (ayj) [inline galley/pkg/config/analysis/analyzers/analyzers_test.go]: You could use StopTimer and StartTimer to pause the test timer during the initialization portion of this function (see ://golang.org/pkg/testing/#B.StopTimer).

Comment (sushicw) [reply galley/pkg/config/analysis/analyzers/analyzers_test.go]: Went ahead and split this function so I could use `b.ResetTimer()`

**Pull request merged**